### ISSUE TITLE
Correct way of using fetchMembers

### ISSUE THREAD

**Issue by Catnippi** [Merubokkusu/Discord-S.C.U.M#37]
Hi, I've been using this example to experiment with fetching discords ranging from 1,000 members to 30,000 members. 
https://github.com/Merubokkusu/Discord-S.C.U.M/blob/master/examples/gettingGuildMembers.py

However I have noticed after the flag of `bot.gateway.finishedMemberFetching(guild_id):` is valid it will only return 6,000 - 7,000. 
I thought I could solve this issue by increasing the startIndex by 50 each search until it would reach an expected value I manually defined. 

ie
> 	bot.gateway.fetchMembers(guild_id, channel_id, keep="all", wait=2, method="no overlap", startIndex=index) #get all user attributes, wait 1 second between requests

and once the fetch is finished index would increase by 50.

However my findings are very unreliable or just not working at all?

> Starting Search at 0 seconds with startIndex 0
> 5780 members fetched
> Starting Search 2 at 224 seconds with startIndex 60
> 82 members fetched
> Starting Search 3 at 314 seconds with startIndex 120
> 81 members fetched
> done

My code is almost identical to the example listed above however I added a loop to call the get_members function after completion - this leads to the question, what is the appropriate way to fully fetch a server members?
_________________________
arandomnewaccount:
it looks like what's happening here is the guild you're experimenting on has 6000-7000 fetchable members (note, # of members your client can see is usually less than the # of total members). To calculate the # of fetchable members, do this: https://github.com/Merubokkusu/Discord-S.C.U.M/blob/master/docs/fetchingGuildMembers.md#calculating--of-fetchable-members

As for your looping, any start index will update the first member group (up to 100 members) since the request for lazy requesting is always for the 0,99 range (explains why you're getting 81-82 members fetched) (https://github.com/Merubokkusu/Discord-S.C.U.M/blob/5a4b09d54df30e7f4d77eaf10f0fa2b32c83e39e/discum/gateway/guild/combo.py#L90). To check if this is the case, see if the size of the member list is increasing each time you're increasing the start index (test it with ```bot.gateway.fetchMembers(..., reset=False)```. If the member list size doesn't increase, then the already-fetched members are just getting updated.

Anyways, I can do some tests later today to double check that it's working smoothly.

edit: 
just double checked on my end on a few guilds (both small and large) and it's working smoothly.

edit:
Closing this issue in 2 days if there's no response. 
tldr of what I said above: 
The examples for fetching members given in this repo are correct. No, you should not be using a loop with increasing start indexes (start index is only actually useful if discord closes your connection to the gateway while you're still fetching the member list).
